server/src/app/api/routes/inference.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (C) 2021, Pyronear contributors.
+
+# This program is licensed under the GNU Affero General Public License version 3.
+# See LICENSE or go to <https://www.gnu.org/licenses/agpl-3.0.txt> for full license details.
+
+
+from fastapi import APIRouter, BackgroundTasks
+from fastapi import File, UploadFile
+
+from app.api.schemas import InferenceOut
+
+router = APIRouter()
+
+
+def predict_and_alert(file):
+    """ predict smoke from an image and if yes raise an alert """
+    # TODO : use inference class and pyro-api client to raise an alert if needed
+    return {"comment": "set fire to the rain please"}
+
+
+@router.post("/", response_model=InferenceOut, status_code=201, summary="Send img from a device to predict smoke")
+async def inference(background_tasks: BackgroundTasks,
+                    file: UploadFile = File(...)
+                    ):
+    """
+    Send img from a device based on the given information in order to predict smoke
+
+    Below, click on "Schema" for more detailed information about arguments
+    or "Example Value" to get a concrete idea of arguments
+    """
+    # some async operation could happen here
+
+    # We might need a backround task here for inference ?
```

Comment:
Could you elaborate?
Are you implying that we don't need the output? If not, I don't see why background task would help

Reply:
The client (here the pi camera that send the image) does not have to wait for the inference & alerting step to continue, we can juste return something like  "Accepted" and the process run in background. That is why I think a [backround task](https://fastapi.tiangolo.com/tutorial/background-tasks/#background-tasks) could be relevant but I may be going the wrong way !

Reply:
Well it depends but if the central pi does have a camera, we might want to raise an alert right after analyzing (if it detects something). So are we doing this only for cameras --> central interactions?

Reply:
If we consider a camera on the central pi, it could send image to the local webserver, and it will work like any other cameras. 
But I might have misunderstood something. Concretely, the background task is launch just after the response of the request right ? So right after sending it, it just let the device that send image not to wait during the inference process.

Reply:
Well yes it runs afterwards but it's asynchronous so the request won't wait for the background task to be executed! It is a viable option sure, depends on what you prefer

Reply:
I would to try this way first and refactor later if needed !